app/src/main/java/com/mapbox/maps/testapp/examples/OfflineActivity.kt

```diff
@@ -0,0 +1,329 @@
+import kotlinx.android.synthetic.main.activity_offline.*
+import kotlinx.android.synthetic.main.activity_offline.recycler
+import java.util.ArrayList
+
+/**
+ * Example app that downloads an offline region and when succeeded
+ * shows a button to load a map at the offline region definition.
+ */
+class OfflineActivity : AppCompatActivity() {
+
+  private lateinit var offlineManager: OfflineManager
+  private var mapView: MapView? = null
+  private lateinit var handler: Handler
+  private lateinit var offlineLogsAdapter: OfflineLogsAdapter
+
+  override fun onCreate(savedInstanceState: Bundle?) {
+    super.onCreate(savedInstanceState)
+    setContentView(R.layout.activity_offline)
+    handler = Handler()
+
+    recycler.layoutManager = LinearLayoutManager(this)
+    offlineLogsAdapter = OfflineLogsAdapter()
+    recycler.adapter = offlineLogsAdapter
+
+    prepareDownloadButton()
+  }
+
+  private fun prepareDownloadButton() {
+    updateButton("DOWNLOAD") {
+      downloadOfflineRegion()
+    }
+  }
+
+  private fun prepareCancelButton(cancelables: List<Cancelable>) {
+    updateButton("CANCEL DOWNLOAD") {
+      cancelables.forEach {
+        it.cancel()
+      }
+      prepareDownloadButton()
+    }
+  }
+
+  private fun prepareViewMapButton() {
+    handler.post {
+      updateButton("VIEW MAP") {
+        // create mapView
+        mapView = MapView(this@OfflineActivity).also { mapview ->
+          val mapboxMap = mapview.getMapboxMap()
+          mapboxMap.setCamera(CameraOptions.Builder().zoom(ZOOM).center(TOKYO).build())
+          mapboxMap.loadStyleUri(Style.OUTDOORS)
+        }
+        container.addView(mapView)
+
+        mapView?.onStart()
+        prepareShowDownloadedRegionButton()
+      }
+    }
+  }
+
+  private fun prepareShowDownloadedRegionButton() {
+    updateButton("SHOW DOWNLOADED REGIONS") {
+      showDownloadedRegions()
+      prepareRemoveOfflineRegionButton()
+    }
+  }
+
+  private fun prepareRemoveOfflineRegionButton() {
+    updateButton("REMOVE DOWNLOADED REGIONS") {
+      removeOfflineRegions()
+      showDownloadedRegions()
+      container.removeAllViews()
+      prepareDownloadButton()
+    }
+  }
+
+  private fun updateButton(text: String, listener: View.OnClickListener) {
+    button.text = text
+    button.setOnClickListener(listener)
+  }
+
+  private fun downloadOfflineRegion() {
+    offlineManager = OfflineManager(MapInitOptions.getDefaultResourceOptions(this))
+
+    // 1. Create style package with loadStylePack() call.
+    val stylePackCancelable = offlineManager.loadStylePack(
+      Style.OUTDOORS,
+      StylePackLoadOptions.Builder()
+        .glyphsRasterizationMode(GlyphsRasterizationMode.IDEOGRAPHS_RASTERIZED_LOCALLY)
+        .metadata(Value(STYLE_PACK_METADATA))
+        .build(),
+      { progress ->
+        updateStylePackDownloadProgress(
+          (progress.completedResourceCount * 100 / progress.requiredResourceCount).toInt(),
```

Comment:
Please add check here in case requiredResourceCount == 0

Reply:
reworked the progress bar update, will not use the divide operation anymore.